### PR TITLE
FIX bad performance with event income and expense

### PR DESCRIPTION
**Pull request by ecino**

### CODE REVIEW

File: crm_compassion/models/account_invoice_line.py [hunk 1]
```diff
@@ -21,6 +21,10 @@ class AccountInvoiceLine(models.Model):
     currency_id = fields.Many2one(
         'res.currency', 'Currency', related='invoice_id.currency_id',
         store=True)
+    event_id = fields.Many2one(
+        'crm.event.compassion', 'Event',
+        related='account_analytic_id.event_id', store=True, readonly=True
+    )
```
Comment (eicher31) [inline crm_compassion/models/account_invoice_line.py:27]: Why use store=True in this case?

File: gift_compassion/models/sponsorship_gift.py [hunk 2]
```diff
@@ -187,11 +187,27 @@ def _compute_usd(self):
     @api.model
     def create(self, vals):
         """ Try to find existing gifts before creating a new one. """
+        gift_date = vals.get('gift_date')
+        if not gift_date:
+            invl = self.env['account.invoice.line']
+            dates = []
+            default = fields.Date.today()
+            for invl_write in vals.get('invoice_line_ids', [[3]]):
+                if invl_write[0] == 0:
```
Comment (eicher31) [inline gift_compassion/models/sponsorship_gift.py:196]: Why [[3]]?

**Review by eicher31 — APPROVED** [review]

File: crm_compassion/models/account_invoice_line.py (hunk 1 above)
Comment (ecino) [reply crm_compassion/models/account_invoice_line.py:27]: To put it in the database and be able to search from the object directly

File: gift_compassion/models/sponsorship_gift.py (hunk 2 above)
Comment (ecino) [reply gift_compassion/models/sponsorship_gift.py:196]: So that it goes in the else case when no values are set.

**Pull request merged**